### PR TITLE
FixerConfigurationResolver - remove mapRootConfigurationTo

### PR DESCRIPTION
**Pull request by keradus**

### CODE REVIEW

File: src/FixerConfiguration/FixerConfigurationResolverRootless.php [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    {
+        return $this->resolver->getOptions();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function resolve(array $options)
+    {
+        if (null !== $this->root && !array_key_exists($this->root, $options) && count($options)) {
```
Comment (julienfalque) [inline src/FixerConfiguration/FixerConfigurationResolverRootless.php]: `$this->root` should never be `null`.

File: src/FixerConfiguration/FixerConfigurationResolverRootless.php [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+ *
+ * @deprecated will be removed in 3.0
+ */
+final class FixerConfigurationResolverRootless implements FixerConfigurationResolverInterface
+{
+    /**
+     * @var FixerConfigurationResolverInterface
+     */
+    private $resolver;
+    /**
```
Comment (julienfalque) [inline src/FixerConfiguration/FixerConfigurationResolverRootless.php]: Missing empty line.

File: src/FixerConfiguration/FixerConfigurationResolverRootless.php (hunk 1 above)
Comment (julienfalque) [inline src/FixerConfiguration/FixerConfigurationResolverRootless.php]: `$this->root` should never be `null`.

File: src/FixerConfiguration/FixerConfigurationResolverRootless.php [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+     */
+    public function __construct($root, $options)
+    {
+        $this->resolver = new FixerConfigurationResolver($options);
+
+        $names = array_map(
+            function (FixerOptionInterface $option) {
+                return $option->getName();
+            },
+            $this->resolver->getOptions()
```
Comment (julienfalque) [inline src/FixerConfiguration/FixerConfigurationResolverRootless.php:46]: Why not use `$options` directly?

**Comment by keradus** [comment]
haha, i already fix it and then forgot to push :D

File: src/FixerConfiguration/FixerConfigurationResolverRootless.php [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    }
+
+    /**
+     * @param FixerOptionInterface $option
+     *
+     * @throws \LogicException when the option is already defined
+     *
+     * @return $this
+     */
+    private function addOption(FixerOptionInterface $option)
```
Comment (julienfalque) [inline src/FixerConfiguration/FixerConfigurationResolverRootless.php]: Not used.

Comment (keradus) [reply src/FixerConfiguration/FixerConfigurationResolverRootless.php]: not there as well

File: src/FixerConfiguration/FixerConfigurationResolverRootless.php (hunk 3 above)
Comment (keradus) [reply src/FixerConfiguration/FixerConfigurationResolverRootless.php:46]: $options is iterable. iterable may not be rewindable

**Pull request merged**